### PR TITLE
Added dockerfile for web and updated docker-compose

### PR DESCRIPTION
**Pull request by ShivamKumar14**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ x] This PR modifies application configuration/environment variables

## How did you test it?

I manually tested the changes by spinning up the Docker Compose environment and verifying that the web client and control center services are properly integrated with the existing services. All services were able to communicate with each other as expected.

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Dockerfile.web-client [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Use official Node.js image as the base image for web client
+FROM node:14 as web-client-builder
```
Comment (SanchithHegde) [inline Dockerfile.web-client:2]: Could we use a more recent version of `node`?

File: Dockerfile.web-client [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+WORKDIR /app
+
+# Copy package.json and package-lock.json to the working directory
+COPY package*.json ./
+
+# Install dependencies
+RUN npm install
+
+# Copy the rest of the application code to the working directory
+COPY . .
```
Comment (SanchithHegde) [inline Dockerfile.web-client:14]: The source of the web client and demo app can be found at https://github.com/juspay/hyperswitch-web.

So you'd have to either add this Dockerfile there, or add steps in this Dockerfile to clone the repository.

File: docker-compose.yml [hunk 3]
```diff
@@ -122,6 +122,28 @@ services:
     labels:
       logs: "promtail"
 
+    web-client:
+    # Configuration for the web client service
+    image: <web_client_image>:<tag>
```
Comment (SanchithHegde) [inline docker-compose.yml:127]: As it stands, we don't have Docker images for the web client on Docker Hub, we'll have to build and start the service instead.

File: Dockerfile.web-client (hunk 1 above)
Comment (ShivamKumar14) [reply Dockerfile.web-client]: yes, sure. done

**Comment by SanchithHegde** [comment]
Closing this in favor of #4197.

**Pull request closed without merging**